### PR TITLE
add completion script generator for bash & fish

### PR DESCRIPTION
**Pull request by 1ntEgr8**
This adds the fish completion scripts referenced in #347 

I began experimenting with bash completion. I took a look at [cobra's](https://github.com/spf13/cobra/blob/master/bash_completions.go) implementation, and it doesn't look as straightforward as the fish shell one.

### CODE REVIEW

**Comment by 1ntEgr8** [comment]
update: added completions for bash!

**Comment by danteissaias** [comment]
This looks great! However is it possible to detect if you're running `dev-rome` and generating completions accordingly.

**Comment by sebmck** [comment]
@1ntEgr8 Awesome! The usage of template literals that aren't indented break the flow of the file. Could you use the `dedent` tagged template literal?

```
import {dedent} from '@romejs/string-utils'`
dedent`code here`;
```

This will make it so that indentation is removed from the beginning of all lines and makes the code more readable.

**Comment by 1ntEgr8** [comment]
@danteissaias to generate completions for dev-rome, i currently read if the ROME_DEV env variable is set to '1'. 

So for example, if you want dev-rome completions for fish, you can run 
```fish
env ROME_DEV=1 ./scripts/dev-rome --generate-autocomplete fish
```

File: result [hunk 1]
```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env bash
```
Comment (sebmck) [inline result:1]: I don't think this file was meant to be committed.

File: packages/@romejs/cli-flags/Parser.ts [hunk 2]
```diff
@@ -599,30 +606,146 @@ export default class Parser<T> {
...
     }
 
-    // this.declaredFlags contains flag information, ignore the keys as they will have command suffixes
-    // utilize `command` for the owned command and `name` for the actual flag name
-
-    // this.commands contains command information
-
-    // reporter.logAllNoMarkup to output to stdout
-    reporter;
+    const prg = process.env.ROME_DEV ? 'dev-rome' : 'rome';
```
Comment (sebmck) [inline packages/@romejs/cli-flags/Parser.ts:617]: We want the `cli-flags` package to be usable by other tools.

```suggestion
    const prg = this.options.programName;
```

If we really want to support `dev-rome` then we should set `programName` in `packages/@romejs/cli/cli.ts` and have the `ROME_DEV` environment variable set by the `dev-rome` script.

**Comment by sebmck** [comment]
Looks good apart from a couple of nits! Let me know if you have any questions. I'm available in the Discord to help too.

**Comment by sebmck** [comment]
Awesome! Looks good to me. Thanks @1ntEgr8!

**Pull request merged**